Sources/Commands/SwiftTool.swift

```diff
@@ -74,39 +74,55 @@ private class ToolWorkspaceDelegate: WorkspaceDelegate {
         self.observabilityScope = observabilityScope
     }
 
-    func fetchingWillBegin(repository: String, fetchDetails: RepositoryManager.FetchDetails?) {
+    func willFetchPackage(package: String, fetchDetails: PackageFetchDetails) {
         queue.async {
-            self.outputStream <<< "Fetching \(repository)"
-            if let fetchDetails = fetchDetails {
-                if fetchDetails.fromCache {
-                    self.outputStream <<< " from cache"
-                }
+            self.outputStream <<< "Fetching \(package)"
+            if fetchDetails.fromCache {
+                self.outputStream <<< " from cache"
             }
             self.outputStream <<< "\n"
             self.outputStream.flush()
         }
     }
 
-    func fetchingDidFinish(repository: String, fetchDetails: RepositoryManager.FetchDetails?, diagnostic: Basics.Diagnostic?, duration: DispatchTimeInterval) {
+    func didFetchPackage(package: String, result: Result<PackageFetchDetails, Error>, duration: DispatchTimeInterval) {
         queue.async {
             if self.observabilityScope.errorsReported {
                 self.fetchAnimation.clear()
             }
 
-            let step = self.fetchProgress.values.reduce(0) { $0 + $1.objectsFetched }
-            let total = self.fetchProgress.values.reduce(0) { $0 + $1.totalObjectsToFetch }
+            let progress = self.fetchProgress.values.reduce(0) { $0 + $1.progress }
+            let total = self.fetchProgress.values.reduce(0) { $0 + $1.total }
 
-            if step == total && !self.fetchProgress.isEmpty {
+            if progress == total && !self.fetchProgress.isEmpty {
                 self.fetchAnimation.complete(success: true)
                 self.fetchProgress.removeAll()
             }
 
-            self.outputStream <<< "Fetched \(repository) (\(duration.descriptionInSeconds))"
+            self.outputStream <<< "Fetched \(package) (\(duration.descriptionInSeconds))"
             self.outputStream <<< "\n"
             self.outputStream.flush()
         }
     }
 
+    func fetchingPackage(package: String, progress: Int64, total: Int64?) {
+        queue.async {
+            self.fetchProgress[package] = FetchProgress(
+                progress: progress,
+                total: total ?? progress
+            )
+
+            let progress = self.fetchProgress.values.reduce(0) { $0 + $1.progress }
+            let total = self.fetchProgress.values.reduce(0) { $0 + $1.total }
+
+            self.fetchAnimation.update(
+                step: progress > Int.max ? Int.max : Int(progress),
+                total: total > Int.max ? Int.max : Int(total),
+                text: "Fetching \(package)"
```

Comment:
For packages that do come from repositories, will there be a way to still see that URL?  Based on how this is called it looks as if this is always the string form of the package identity.

Reply:
this PR does not change the behavior for repository based dependencies, ie they still pass the url